Automattic/jetpack
# Instant Search: Migrate settings from widgets into the customizer

#### Changes proposed in this Pull Request:
* Updated Jetpack Search customizer section with two new options: Default sort and the ability to hide the sort selector.
<img width="548" alt="Customize__Instant_Search_Test_–_Just_another_WordPress_site" src="https://user-images.githubusercontent.com/4044428/81610631-129f5c80-9397-11ea-8555-8ca6b62a9b54.png">

* Removed unused Jetpack Search sort widget option when Instant Search is enabled.
> <img width="297" alt="Screen Shot 2020-05-27 at 1 38 14 PM" src="https://user-images.githubusercontent.com/4044428/83064910-d10de300-a01f-11ea-8e26-d84ac331ddc6.png">

* Adds support for post types whitelist in the Jetpack Search widget configuration in the Customizer.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Yes. This change adds the following for Jetpack Instant Search:
- Ability to set a default sort for search results.
- Ability to hide the sort selector.
- Support for whitelisting post types in Instant Search.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to a site with Jetpack Instant Search enabled.
* Add a Jetpack Search widget to your site's footer or sidebar via the customizer. Ensure that the widget title and filters remain configurable.
* Add a Jetpack Search widget to the Jetpack Search sidebar via the customizer. Ensure that only the widget filters remain configurable.

* Try limiting the widget's post type whitelist to only allow for a certain post type. Ensure that summoning the overlay using this widget's search input applies the necessary post type filter.

* Navigate to the Jetpack Search section of the customizer. Ensure that the default sort defaults to `Relevance` and that the `Show Sort Selector` defaults to being checked.
* Try changing the default sort; this should reload the iframe preview. Ensure that the default sort has changed by verifying that your newly selected default sort has been pre-selected by the sort selector.
<img width="686" alt="Screen Shot 2020-05-11 at 3 13 39 PM" src="https://user-images.githubusercontent.com/4044428/81612412-036dde00-939a-11ea-9398-9915a1721762.png">

* Try unchecking the `Show Sort Selector` checkbox. This should hide the sort selector without refreshing the page.

<img width="686" alt="Screen Shot 2020-05-11 at 3 14 36 PM" src="https://user-images.githubusercontent.com/4044428/81612478-239d9d00-939a-11ea-86eb-a27c069dd344.png">

#### Proposed changelog entry for your changes:
* Adds support for setting a default sort order for Jetpack Instant Search results.
* Adds support for hiding the sort selector for Jetpack Instant Search.
